app/src/features/surveys/components/SurveyBlockSection.tsx

```diff
@@ -0,0 +1,154 @@
+import { mdiPencilOutline, mdiPlus, mdiTrashCanOutline } from '@mdi/js';
+import Icon from '@mdi/react';
+import MoreVertIcon from '@mui/icons-material/MoreVert';
+import { ListItemIcon, Menu, MenuItem, MenuProps } from '@mui/material';
+import Box from '@mui/material/Box';
+import Button from '@mui/material/Button';
+import Card from '@mui/material/Card';
+import CardHeader from '@mui/material/CardHeader';
+import IconButton from '@mui/material/IconButton';
+import { useFormikContext } from 'formik';
+import { ICreateSurveyRequest } from 'interfaces/useSurveyApi.interface';
+import React, { useState } from 'react';
+import yup from 'utils/YupSchema';
+import CreateSurveyBlockDialog from './CreateSurveyBlockDialog';
+import EditSurveyBlockDialog from './EditSurveyBlockDialog';
+
+interface ISurveyBlockFormProps {
+  name: string; // the name of the formik field value
+}
+
+export const SurveyBlockInitialValues = {
+  blocks: []
+};
+
+// Form validation for Block Item
+export const BlockYupSchema = yup.object({
+  name: yup.string().required(),
+  description: yup.string().required()
+});
+
+export const SurveyBlockYupSchema = yup.array(BlockYupSchema);
+
+export interface IEditBlock {
+  index: number;
+  block: {
+    survey_block_id: number | null;
+    name: string;
+    description: string;
+  };
+}
+
+const SurveyBlockSection: React.FC<ISurveyBlockFormProps> = (props) => {
+  const { name } = props;
+  const [isCreateModalOpen, setIsCreateModalOpen] = useState(false);
+  const [isEditModalOpen, setIsEditModalOpen] = useState(false);
+  const [anchorEl, setAnchorEl] = useState<MenuProps['anchorEl']>(null);
+  const [editData, setEditData] = useState<IEditBlock | undefined>(undefined);
+
+  const formikProps = useFormikContext<ICreateSurveyRequest>();
+  const { values, handleSubmit, setFieldValue } = formikProps;
+
+  const handleMenuClick = (event: React.MouseEvent<HTMLDivElement, MouseEvent>, index: number) => {
+    setAnchorEl(event.currentTarget);
+    setEditData({ index: index, block: values.blocks[index] });
+  };
+
+  const handleDelete = () => {
+    if (editData) {
+      const data = values.blocks;
+      data.splice(editData.index, 1);
+      setFieldValue(name, data);
+    }
+    setAnchorEl(null);
+  };
+
+  return (
+    <>
+      {/* CREATE BLOCK DIALOG */}
```

Comment:
Nit: Remove comments

Reply:
the components are properly named but I think the comments give good space between everything else